src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+
+type Props = {
+  todos: Todo[];
+  removeTodo: (id: number) => Promise<void>;
+  tempTodo: Todo | null;
+  isLoading?: boolean | string;
+  renameTodo: (id: number, newTitle: string) => Promise<void>;
+  toggleTodo: (id: number) => void;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  removeTodo,
+  tempTodo,
+  isLoading,
+  renameTodo,
+  toggleTodo,
+}) => {
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+  const [updatingTodoId, setUpdatingTodoId] = useState<number | null>(null);
+  const [togglingTodoId, setTogglingTodoId] = useState<number | null>(null);
+
+  const handleRemoveTodo = (id: number) => {
+    setDeletingTodoId(id);
+
+    return removeTodo(id);
+  };
+
+  const handleUpdateTodo = (id: number, newTitle: string): Promise<void> => {
+    setUpdatingTodoId(id);
+
+    return renameTodo(id, newTitle);
+  };
+
+  const handleToggleTodo = (id: number) => {
+    setTogglingTodoId(id);
+    toggleTodo(id);
+  };
+
+  const allCompleted = todos.every(todo => todo.completed);
```

Comment:
```suggestion
  const isAllCompleted = todos.every(todo => todo.completed);
```
it's a boolean value. Same in the App.tsx